apiserver/facades/client/cloud/cloud_test.go

```diff
@@ -268,109 +313,149 @@ func createAddCloudParam(cloudType string) params.AddCloudArgs {
-		errors.New("kaboom"), // Cloud
-	)
+	adminTag := names.NewUserTag("admin")
+	defer s.setup(c, adminTag).Finish()
+
+	backend := s.backend.EXPECT()
+	backend.ControllerInfo().Return(&state.ControllerInfo{CloudName: "kaboom"}, nil)
+	backend.Cloud("kaboom").Return(jujucloud.Cloud{}, errors.New("kaboom"))
+
 	err := s.api.AddCloud(createAddCloudParam(""))
 	c.Assert(err, gc.ErrorMatches, "kaboom")
-	s.backend.CheckCallNames(c, "ControllerInfo", "Cloud")
 }
 
 func (s *cloudSuite) TestAddCloudK8sForceIrrelevant(c *gc.C) {
-	s.backend.controllerCfg = controller.Config{
-		"features": []interface{}{"multi-cloud"},
+	adminTag := names.NewUserTag("admin")
+	defer s.setup(c, adminTag).Finish()
+
+	cloud := jujucloud.Cloud{
+		Name:      "newcloudname",
+		Type:      string(k8sconstants.CAASProviderType),
+		AuthTypes: []jujucloud.AuthType{jujucloud.EmptyAuthType, jujucloud.UserPassAuthType},
+		Endpoint:  "fake-endpoint",
+		Regions:   []jujucloud.Region{{Name: "nether", Endpoint: "nether-endpoint"}},
 	}
+
+	backend := s.backend.EXPECT()
+	backend.AddCloud(cloud, adminTag.Name()).Return(nil).Times(2)
+
 	addCloudArg := createAddCloudParam(string(k8sconstants.CAASProviderType))
+
 	add := func() {
 		err := s.api.AddCloud(addCloudArg)
 		c.Assert(err, jc.ErrorIsNil)
-		s.backend.CheckCalls(c, []gitjujutesting.StubCall{
-			{"AddCloud", []interface{}{cloud.Cloud{
-				Name:      "newcloudname",
-				Type:      string(k8sconstants.CAASProviderType),
-				AuthTypes: []cloud.AuthType{cloud.EmptyAuthType, cloud.UserPassAuthType},
-				Endpoint:  "fake-endpoint",
-				Regions:   []cloud.Region{{Name: "nether", Endpoint: "nether-endpoint"}},
-			}, "admin"}},
-		})
 	}
 	add()
-
 	force := true
-	s.backend.ResetCalls()
 	addCloudArg.Force = &force
 	add()
 }
 
 func (s *cloudSuite) TestAddCloudNoRegion(c *gc.C) {
-	s.backend.cloud.Type = "maas"
+	adminTag := names.NewUserTag("admin")
+	defer s.setup(c, adminTag).Finish()
+
+	newCloud := jujucloud.Cloud{
+		Name:      "newcloudname",
+		Type:      "maas",
+		AuthTypes: []jujucloud.AuthType{jujucloud.EmptyAuthType, jujucloud.UserPassAuthType},
+		Endpoint:  "fake-endpoint",
+		Regions: []jujucloud.Region{{
+			Name: "default",
+		}},
+	}
+
+	cloud := jujucloud.Cloud{
+		Name: "newcloudname",
+		Type: "maas",
+	}
+
+	backend := s.backend.EXPECT()
+	backend.ControllerInfo().Return(&state.ControllerInfo{CloudName: "newcloudname"}, nil)
+	backend.Cloud("newcloudname").Return(cloud, nil)
+	backend.AddCloud(newCloud, adminTag.Name()).Return(nil)
 	paramsCloud := params.AddCloudArgs{
 		Name: "newcloudname",
 		Cloud: params.Cloud{
 			Type:      "maas",
 			AuthTypes: []string{"empty", "userpass"},
 			Endpoint:  "fake-endpoint",
 		}}
-	s.backend.addCloudF = func(cloud cloud.Cloud, user string) error {
-		c.Assert(cloud.Regions, gc.HasLen, 1)
-		return nil
-	}
+
+	c.Assert(newCloud.Regions, gc.HasLen, 1)
```

Comment:
Drop this, since it's hard coded to be true above

Reply:
Done